server/app/main.py

```diff
@@ -41,70 +40,86 @@ async def get_status(request):
+
+
 async def get_measurements(request):
     """Return measurements sorted chronologically, optionally filtered."""
+    request = await validation.validate(request, validation.GetMeasurementsRequest)
 
-    # TODO Simplify this part somehow so that we don't have to duplicate it
-    try:
-        # TODO Use one model for body/query/... (with inheritance?)
-        request = validation.GetMeasurementsRequest(**request.query_params)
-    except pydantic.ValidationError:
-        # TODO Include specific pydantic/attrs error message
-        logger.warning("GET /measurements: InvalidSyntaxError")
-        raise errors.InvalidSyntaxError()
-
-    # Define default columns and conditions
-    # TODO move this into validation, and set some sensible defaults, e.g. limit=64
+    # Define the returned columns
     columns = [
-        column
-        for column in MEASUREMENTS.columns
-        if column not in [MEASUREMENTS.columns.receipt_timestamp]
+        MEAS.c.sensor_identifier,
+        MEAS.c.measurement_timestamp,
+        *[sa.column(value_identifier) for value_identifier in request.query.values],
     ]
-    conditions = []
-
-    # Build customized database query from query parameters
-    if request.nodes is not None:
-        conditions.append(
-            sa.or_(
-                *[
-                    MEASUREMENTS.columns.node_identifier == node_identifier
-                    for node_identifier in request.nodes
-                ]
-            )
-        )
-    if request.values is not None:
-        columns = [
-            MEASUREMENTS.columns.node_identifier,
-            MEASUREMENTS.columns.measurement_timestamp,
-            *[
-                sa.column(value)
-                for value_identifier in request.values
-                if value_identifier
-                in set(MEASUREMENTS.columns.keys())
-                - {"node_identifier", "measurement_timestamp", "receipt_timestamp"}
-            ],
-        ]
-    if request.start_timestamp is not None:
+    # Define filter by sensor_identifier
+    conditions = [
+        sa.or_(
+            MEAS.c.sensor_identifier == sensor_identifier
+            for sensor_identifier in request.query.sensors
+        ),
+    ]
+    # Define filter by measurement_timestamp
+    if request.query.start_timestamp is not None:
         conditions.append(
-            MEASUREMENTS.columns.measurement_timestamp >= int(request.start_timestamp)
+            MEAS.c.measurement_timestamp >= int(request.query.start_timestamp)
         )
-    if request.end_timestamp is not None:
+    if request.query.end_timestamp is not None:
         conditions.append(
-            MEASUREMENTS.columns.measurement_timestamp < int(request.end_timestamp)
+            MEAS.c.measurement_timestamp < int(request.query.end_timestamp)
         )
-
+    # Build query
+    query = (
+        sa.select(columns)
+        .select_from(MEAS)
+        .where(sa.and_(*conditions))
```

Comment:
`conditions` could have a better name, so this part is more readable.

Reply:
Do you have a proposition?

Reply:
Something like `query_includes_sensor`? It is kind of hard to understand what this ...

```python
and_(
    or_(
        MEAS.c.sensor_identifier == sensor_identifier
        for sensor_identifier in request.query.sensors
    )
)
```

... is doing. I know what it does, but the format is very cryptic.

Reply:
I'm still trying around how to write SQLAlchemy Core queries in the most readable way, but I'm trying to structure them in a similar way.

I want to use a `conditions` variable for every query, and it should contain all conditions in a SQL `WHERE` clause. `query_includes_sensor` is too specific. In the case of `GET /measurements` `conditions` can look like this:

```python
and_(
    or_(
        MEAS.c.sensor_identifier == "sensor-123"
        MEAS.c.sensor_identifier == "sensor-strawberry"
    ),
    MEAS.c.measurement_timestamp >= 100,
    MEAS.c.measurement_timestamp < 99999,
)
```

For `GET /sensors` you're right for the moment though.

Reply:
Totally agreed. Having one `conditions`, `query`, ... variable per db request is a good structure.

The SQL Alchemy notation stuff takes a while to get into. But by now I am used to it. Everyone who works with the code itself will have to read into the sqlalchemy-use anyway.